File: GLLara/GLLAmbientLight.m
//
//  GLLAmbientLight.m
//  GLLara
//
//  Created by Torsten Kammer on 09.09.12.
//  Copyright (c) 2012 Torsten Kammer. All rights reserved.
//

#import "GLLAmbientLight.h"


@implementation GLLAmbientLight

@dynamic color;
@dynamic index;

#pragma mark - Source list item

- (BOOL)isSourceListHeader
{
	return NO;
}
- (NSString *)sourceListDisplayName
{
	return NSLocalizedString(@"Ambient", @"source list: ambient light name");
}
- (BOOL)isLeafInSourceList
{
	return YES;
}
- (NSUInteger)countOfSourceListChildren
{
	return 0;
}
- (id)objectInSourceListChildrenAtIndex:(NSUInteger)index;
{
	return nil;
}


@end

File: GLLara/GLLAmbientLight.swift
//
//  GLLAmbientLight+CoreDataClass.swift
//  GLLara
//
//  Created by Torsten Kammer on 05.05.24.
//  Copyright © 2024 Torsten Kammer. All rights reserved.
//
//

import Foundation
import CoreData

/**
 * @abstract A scene's ambient light information.
 * @discussion Each scene will have only one. The index is only for sorting in
 * the UI.
 */
@objc(GLLAmbientLight)
public class GLLAmbientLight: NSManagedObject {
    @NSManaged public var color: NSColor!
    @NSManaged public var index: Int64
}
